### DISCUSSION TITLE
style.json text-size across zoom using categories

### DISCUSSION THREAD

**Discussion by mar0321** [maplibre/maplibre-gl-js#2062] [Q&A]
Hi i would like to ask if there is a possibility to have one layer in style.json e.g. cities:
in data i have collum caled cat which has values for size of cities

is it possible to set text-size with property cat =1, than text-size = e.g 10 but i would like set these size categories different in each zoom.

Now i am able to set different text size for each zoom level but i also need to add fillter category so for different city category will be different text-size in one layer...

Right now i have like 10 city-label layers which is annoying for engine to render all layers.
Thanks

**Comment by HarelM** [comment]
You might be able to write some expressions, it's a bit complicated and I don't know if it will solve your specific case though.
In any case, this is not really an issue so I'll be converting this to a discussion...
